### PR TITLE
CPS-80: Added Directory.list_marketplaces and Directory.get_marketplace

### PR DESCRIPTION
**Pull request by JaviCerveraIngram**

### CODE REVIEW

File: connect/resources/directory.py [hunk 1]
```diff
@@ -47,6 +48,27 @@ def get_asset(self, asset_id):
         text, code = ApiClient(self._config, 'assets/' + asset_id).get()
         return Asset.deserialize(text)
 
+    def list_marketplaces(self, filters=None):
+        """ List the marketplaces.
+
+        :param dict|Query filters: Filters to pass to the request.
+        :return: A list with the marketplaces that match the given filters.
+        :rtype: list[Asset]
```
Comment (d3rky) [inline connect/resources/directory.py:56]: Missed type here `list[Marketplace]`

File: connect/resources/directory.py [hunk 2]
```diff
@@ -47,6 +48,27 @@ def get_asset(self, asset_id):
...
+        query = self._get_filters_query(filters, False)
+        text, code = ApiClient(self._config, 'marketplaces' + query.compile()).get()
+        return Marketplace.deserialize(text)
+
+    def get_marketplace(self, marketplace_id):
+        """ Returns the marketplace with the given id.
+
+        :param str marketplace_id: The id of the marketplace.
+        :return: The asset with the given id, or ``None`` if such asset does not exist.
+        :rtype: Asset|None
```
Comment (d3rky) [inline connect/resources/directory.py:67]: Missed type, looks like should be `Marketplace|None`

File: connect/resources/directory.py (hunk 1 above)
Comment (JaviCerveraIngram) [reply connect/resources/directory.py:56]: Well spotted. Fixing now.

File: connect/resources/directory.py (hunk 2 above)
Comment (JaviCerveraIngram) [reply connect/resources/directory.py:67]: Fixing...

File: connect/resources/directory.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def list_marketplaces(self, filters=None):
 
         :param dict|Query filters: Filters to pass to the request.
         :return: A list with the marketplaces that match the given filters.
```
Comment (marcserrat) [inline connect/resources/directory.py:55]: List of marketplaces matching given filters

File: connect/resources/directory.py [hunk 4]
```diff
@@ -47,6 +48,27 @@ def get_asset(self, asset_id):
...
+        :param dict|Query filters: Filters to pass to the request.
+        :return: A list with the marketplaces that match the given filters.
+        :rtype: list[Marketplace]
+        """
+        query = self._get_filters_query(filters, False)
+        text, code = ApiClient(self._config, 'marketplaces' + query.compile()).get()
+        return Marketplace.deserialize(text)
+
+    def get_marketplace(self, marketplace_id):
+        """ Returns the marketplace with the given id.
```
Comment (marcserrat) [inline connect/resources/directory.py:63]: Obtains Marketplace object given it's ID

File: connect/resources/directory.py [hunk 5]
```diff
@@ -47,6 +48,27 @@ def get_asset(self, asset_id):
...
+
+        :param str marketplace_id: The id of the marketplace.
+        :return: The asset with the given id, or ``None`` if such asset does not exist.
```
Comment (marcserrat) [inline connect/resources/directory.py:66]: Asset?

File: connect/resources/directory.py (hunk 3 above)
Comment (JaviCerveraIngram) [reply connect/resources/directory.py]: Fixed.

File: connect/resources/directory.py (hunk 4 above)
Comment (JaviCerveraIngram) [reply connect/resources/directory.py]: Fixed.

File: connect/resources/directory.py (hunk 5 above)
Comment (JaviCerveraIngram) [reply connect/resources/directory.py]: Fixed.

**Comment by JaviCerveraIngram** [comment]
> It could be great creating a MarkeplaceResource that inherit from BaseResource and includes it in DirectoryResource. (see ProductResource).
> Then we will have a resource for each object that matches the API and grouped by facades.

But ProductResource does not contain list or get methods for products itself (only for their parameters). The method list_products and get_product are still in the Directory class.

**Comment by martinconstante** [comment]
> > It could be great creating a MarkeplaceResource that inherit from BaseResource and includes it in DirectoryResource. (see ProductResource).
> > Then we will have a resource for each object that matches the API and grouped by facades.
> 
> But ProductResource does not contain list or get methods for products itself (only for their parameters). The methods list_products and get_product are still in the Directory class.

ProductResouce inherits the methods search get create from BaseResource, and implements the specifics methods like the related to parameters. When you want to use the resources of ProductResource you only must include the facade class Directory. The objective is to create a resource class for every business object and group in facade classes like Directory, Fulfillment, Subscription.

**Comment by JaviCerveraIngram** [comment]
> ProductResouce inherits the methods search get create from BaseResource, and implements the specifics methods like the related to parameters. When you want to use the resources of ProductResource you only must include the facade class Directory. The objective is to create a resource class for every business object and group in facade classes like Directory, Fulfillment, Subscription.

Ok, now I get it. I was getting confused because Directory.get_product and Directory.list_products do not use the Product resource to implement the functionality, but they do it directly with ApiClient. So, should I add a MarketplaceResource class and call the resource methods from Directory.list_marketplaces and Directory.get_marketplace?

**Comment by JaviCerveraIngram** [comment]
> Correct!

Thanks, I'll provide the changes ASAP.

**Comment by JaviCerveraIngram** [comment]
Requested changes have been added.

**Pull request merged**